presto-hive/src/main/java/io/prestosql/plugin/hive/security/AccessControlMetadataFactory.java

```diff
@@ -13,9 +13,9 @@
  */
 package io.prestosql.plugin.hive.security;
 
-import io.prestosql.plugin.hive.metastore.SemiTransactionalHiveMetastore;
-
 public interface AccessControlMetadataFactory
 {
-    AccessControlMetadata create(SemiTransactionalHiveMetastore metastore);
+    // TODO get rid of SqlStandardAccessControlMetastore. Possibly replace with something generic
+    // not tied to selected access control implementation. E.g. HiveTransactionHandle.
```

Comment:
Are you sure this TODO is actionable? or it is going to stay with us for ever?

Also, since you're improving the situation here, i don't feel like you need to add a TODO at all

Reply:
I will drop it for now. Not sure if it is easily actionable. Without massive refactor.